src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php

```diff
@@ -55,14 +60,157 @@
 use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductOutOfStockException;
+use PrestaShop\PrestaShop\Core\Search\Filters\CartFilter;
+use PrestaShopBundle\Component\CsvResponse;
 use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
 use PrestaShopBundle\Security\Annotation\AdminSecurity;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
 class CartController extends FrameworkBundleAdminController
 {
+    /**
+     * Shows list of carts
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param CartFilter $filters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, CartFilter $filters)
+    {
+        $cartsKpiFactory = $this->get('prestashop.core.kpi_row.factory.carts');
+        $cartGridFactory = $this->get('prestashop.core.grid.factory.cart');
+        $cartGrid = $cartGridFactory->getGrid($filters);
+
+        return $this->render('@PrestaShop/Admin/Sell/Order/Cart/index.html.twig', [
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'enableSidebar' => true,
+            'layoutHeaderToolbarBtn' => [
+                'add' => [
+                    'href' => $this->generateUrl('admin_carts_export'),
+                    'desc' => $this->trans('Export carts', 'Admin.Orderscustomers.Feature'),
+                    'icon' => 'cloud_download',
+                ],
+            ],
+            'cartGrid' => $this->presentGrid($cartGrid),
+            'cartsKpi' => $cartsKpiFactory->build(),
+        ]);
+    }
+
+    /**
+     * Delete given cart
+     *
+     * @AdminSecurity("is_granted('delete', request.get('_legacy_controller'))", redirectRoute="admin_carts_index")
+     *
+     * @param int $cartId
+     *
+     * @return RedirectResponse
+     */
+    public function deleteCartAction(int $cartId)
+    {
+        try {
+            $this->getCommandBus()->handle(new DeleteCartCommand($cartId));
+            $this->addFlash('success', $this->trans('Successful deletion', 'Admin.Notifications.Success'));
+        } catch (CartException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_carts_index');
+    }
+
+    /**
+     * Provides cart ids from request of bulk action
+     *
+     * @param Request $request
+     *
+     * @return array
+     */
+    private function getBulkCartsFromRequest(Request $request): array
+    {
+        $cartIds = $request->request->get('cart_bulk');
+
+        if (!is_array($cartIds)) {
+            return [];
+        }
+
+        return array_map('intval', $cartIds);
+    }
```

Comment:
you should soon be able to reuse trait https://github.com/PrestaShop/PrestaShop/pull/32447 to get ids instead of adding new method.